packages/netlify-cms-core/src/reducers/collections.js

```diff
@@ -112,9 +112,10 @@ export const selectAllowDeletion = collection =>
   selectors[collection.get('type')].allowDeletion(collection);
 export const selectTemplateName = (collection, slug) =>
   selectors[collection.get('type')].templateName(collection, slug);
-export const selectIdentifier = collection => {
+export const selectIdentifierCandidates = collection => {
   const fieldNames = collection.get('fields').map(field => field.get('name'));
-  return IDENTIFIER_FIELDS.find(id => fieldNames.find(name => name.toLowerCase().trim() === id));
```

Comment:
The only change needed here is doing a `filter` instead of a `find`, and returning the array of field names that occurs in both. We could greatly simplify this by using [`lodash.intersection`](https://lodash.com/docs/4.17.10#intersection). This allows the order of `IDENTIFIER_FIELDS` to dictate which field is used, otherwise this would be a breaking change.